### ISSUE TITLE
Inconsistent karaf output for `addons services`

### ISSUE THREAD

**Issue by dilyanpalauzov** [openhab/openhab-core#4146]
https://github.com/openhab/openhab-core/pull/4057 implements the `openhab:addons services` command in Karaf.

With OH 4.2.0M1 after `openhab-cli clean-cache`, I start openhab.  Then Karaf shows
```
openhab> openhab:addons services
karaf                openHAB Distribution
marketplace          Community Marketplace
json                 Json 3rd Party Add-on Service
jar                  JAR-File add-on service
```
I stop openhab, then start it again and wait until everything is loaded.  Then Karaf prints only:
```
openhab> openhab:addons services
jar                  JAR-File add-on service
```
I repeat the above steps from the beginning (stop oh, clean-cache, start oh, addons services, stop oh, start oh, addons services) and get consistent results: The first time four services are listed, the second time only the jar service is listed.

**Comment by lolodomo** [comment]
I just tried on my running OH and this console command returns only one line for me too:
```
openhab> openhab:addons services
jar                  JAR-File add-on service
```
At the same time, the REST API /addons/services is properly returning 4 items:
```
[
  {
    "id": "jar",
    "name": "JAR-File add-on service",
    "addonTypes": [
      {
        "id": "automation",
        "label": "Automation"
      },
      {
        "id": "binding",
        "label": "Bindings"
      },
      {
        "id": "misc",
        "label": "Misc"
      },
      {
        "id": "persistence",
        "label": "Persistence"
      },
      {
        "id": "transformation",
        "label": "Transformations"
      },
      {
        "id": "ui",
        "label": "User Interfaces"
      },
      {
        "id": "voice",
        "label": "Voice"
      }
    ]
  },
  {
    "id": "karaf",
    "name": "openHAB Distribution",
    "addonTypes": [
      {
        "id": "automation",
        "label": "Automation"
      },
      {
        "id": "binding",
        "label": "Bindings"
      },
      {
        "id": "misc",
        "label": "Misc"
      },
      {
        "id": "persistence",
        "label": "Persistence"
      },
      {
        "id": "transformation",
        "label": "Transformations"
      },
      {
        "id": "ui",
        "label": "User Interfaces"
      },
      {
        "id": "voice",
        "label": "Voice"
      }
    ]
  },
  {
    "id": "marketplace",
    "name": "Community Marketplace",
    "addonTypes": [
      {
        "id": "automation",
        "label": "Automation"
      },
      {
        "id": "binding",
        "label": "Bindings"
      },
      {
        "id": "misc",
        "label": "Misc"
      },
      {
        "id": "persistence",
        "label": "Persistence"
      },
      {
        "id": "transformation",
        "label": "Transformations"
      },
      {
        "id": "ui",
        "label": "User Interfaces"
      },
      {
        "id": "voice",
        "label": "Voice"
      }
    ]
  },
  {
    "id": "json",
    "name": "Json 3rd Party Add-on Service",
    "addonTypes": [
      {
        "id": "automation",
        "label": "Automation"
      },
      {
        "id": "binding",
        "label": "Bindings"
      },
      {
        "id": "misc",
        "label": "Misc"
      },
      {
        "id": "persistence",
        "label": "Persistence"
      },
      {
        "id": "transformation",
        "label": "Transformations"
      },
      {
        "id": "ui",
        "label": "User Interfaces"
      },
      {
        "id": "voice",
        "label": "Voice"
      }
    ]
  }
]
```

Let "s take a look to the console code...

**Comment by lolodomo** [comment]
Code looks good, each AddonService is bound/unbound dynamically to the console service.
https://github.com/openhab/openhab-core/blob/918b4faa3b3254a993774147f798e8bc59580bf6/bundles/org.openhab.core.io.console/src/main/java/org/openhab/core/io/console/internal/extension/AddonConsoleCommandExtension.java#L68

Would require debugging or additional logs to understand what's wrong.

**Comment by lolodomo** [comment]
That is strange, `bindAddonService` is called only once at startup, for the "jar" service.

Looks like an OSGi problem but I do not see what's wrong in the code.

**Comment by lolodomo** [comment]
Maybe `policy = ReferencePolicy.DYNAMIC` is missing, I see it is used elsewhere for `cardinality = ReferenceCardinality.MULTIPLE`

**Comment by lolodomo** [comment]
Setting the policy fixed the issue. I am preparing the PR.

**Comment by dilyanpalauzov** [comment]
Please consider this fix for an 4.2.x release.
